cmd/kube-proxy/app/server_linux_test.go

```diff
@@ -719,3 +721,221 @@ func TestProxyServer_platformSetup(t *testing.T) {
 		})
 	}
 }
+
+type fakeConntracker struct {
+	called []string
+	err    error
+}
+
+func (fc *fakeConntracker) SetMax(ctx context.Context, max int) error {
+	fc.called = append(fc.called, "SetMax")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPEstablishedTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetTCPEstablishedTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPCloseWaitTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetTCPCloseWaitTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetTCPBeLiberal(ctx context.Context, value int) error {
+	fc.called = append(fc.called, "SetTCPBeLiberal")
+	return fc.err
+}
+func (fc *fakeConntracker) SetUDPTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetUDPTimeout")
+	return fc.err
+}
+func (fc *fakeConntracker) SetUDPStreamTimeout(ctx context.Context, seconds int) error {
+	fc.called = append(fc.called, "SetUDPStreamTimeout")
+	return fc.err
+}
+
+func TestSetupConntrack(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	tests := []struct {
+		name         string
+		config       proxyconfigapi.KubeProxyConntrackConfiguration
+		expect       []string
+		conntrackErr error
+		wantErr      bool
+	}{
+		{
+			name: "do nothing if conntrack config is empty",
+			config: proxyconfigapi.KubeProxyConntrackConfiguration{
+				MaxPerCore:            nil,
+				Min:                   nil,
+				TCPEstablishedTimeout: nil,
+				TCPCloseWaitTimeout:   nil,
+				TCPBeLiberal:          false,
+				UDPTimeout:            metav1.Duration{Duration: 0},
+				UDPStreamTimeout:      metav1.Duration{Duration: 0},
+			},
+		},
+		{
+			name: "SetMax is called if conntrack.maxPerCore is specified",
+			config: proxyconfigapi.KubeProxyConntrackConfiguration{
+				MaxPerCore:            ptr.To(int32(12)),
+				Min:                   nil,
+				TCPEstablishedTimeout: nil,
+				TCPCloseWaitTimeout:   nil,
+				TCPBeLiberal:          false,
+				UDPTimeout:            metav1.Duration{Duration: 0},
+				UDPStreamTimeout:      metav1.Duration{Duration: 0},
+			},
+			expect: []string{"SetMax"},
```

Comment:
you could have it append `"SetMax(12)"` rather than just `"SetMax"` so we're sure it's being called with the _right_ value

Reply:
hmm, we can do for the others, but for this one we can not, since the value to be set depends on the num of cpus, see 
```
func getConntrackMax(ctx context.Context, config proxyconfigapi.KubeProxyConntrackConfiguration) (int, error) {
	logger := klog.FromContext(ctx)
	if config.MaxPerCore != nil && *config.MaxPerCore > 0 {
		floor := 0
		if config.Min != nil {
			floor = int(*config.Min)
		}
		scaled := int(*config.MaxPerCore) * detectNumCPU()
```

Reply:
I've added the value to the other methods, but I don't think is worth the effort to complicate the test just for this one